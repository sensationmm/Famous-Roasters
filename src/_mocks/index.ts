import { CatalogueMockError, CatalogueMockMissingData, CatalogueMocks, CatalogueMocksPagination } from './CatalogueMock'
import { ProductMock } from './ProductMock'

export { CatalogueMocks as CatalogueMocks }
export { CatalogueMocksPagination as CatalogueMocksPagination }
export { CatalogueMockError as CatalogueMockError }
export { CatalogueMockMissingData as CatalogueMockMissingData }
export { ProductMock as ProductMock }
