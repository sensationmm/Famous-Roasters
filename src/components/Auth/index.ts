import { Authenticator } from './Authenticator'
import { AuthFormEmail } from './AuthFormEmail'
import { AuthFormItemInput } from './AuthFormItemInput'
import { AuthFormPassword } from './AuthFormPassword'
import { AuthSignIn } from './SignIn'

export { Authenticator as Authenticator }
export { AuthFormEmail as AuthFormEmail }
export { AuthFormItemInput as AuthFormItemInput }
export { AuthFormPassword as AuthFormPassword }
export { AuthSignIn as AuthSignIn }
