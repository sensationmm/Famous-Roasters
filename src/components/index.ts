import { Button, ButtonColor, ButtonEmphasis, ButtonSize } from './Button'
import { Footer } from './Footer'
import { Layout } from './Layout'
import { Navigation } from './Navigation'
import { Typography, TypographySize, TypographyType } from './Typography'

export { Button as Button }
export { ButtonColor as ButtonColor }
export { ButtonEmphasis as ButtonEmphasis }
export { ButtonSize as ButtonSize }
export { Footer as Footer }
export { Layout as Layout }
export { Navigation as Navigation }
export { Typography as Typography }
export { TypographyType as TypographyType }
export { TypographySize as TypographySize }
