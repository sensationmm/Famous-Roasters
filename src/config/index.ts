import { apolloClient } from './apollo'
import i18n from './i18n'

export { i18n as i18n }
export { apolloClient as apolloClient }
