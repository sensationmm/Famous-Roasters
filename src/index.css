@tailwind base;
@tailwind components;
@tailwind utilities;

body {
  margin: 0;
  padding: 0;
  font-family: 'Open Sans', sans-serif;
  font-size: 16px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

body > div > div > main {
  min-height: 50vh;
}

:root {
  --swiper-theme-color: #646575 !important;
}

.video-container {
  position: relative;
  padding-bottom: 56.25%;
  padding-top: 0;
  height: 0;
  overflow: hidden;
}

.video-container iframe,
.video-container object,
.video-container embed {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}