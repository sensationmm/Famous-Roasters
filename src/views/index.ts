import { About } from './About'
import { Account } from './Account'
import { Cart } from './Cart'
import { Catalogue } from './Catalogue'
import { Checkout } from './Checkout'
import { Contact } from './Contact'
import { Error } from './Error'
import { Home } from './Home'
import { OurRoasters } from './OurRoasters'
import { Product } from './Product'
import { TasteFinder } from './TasteFinder'

export { About as About }
export { Account as Account }
export { Cart as Cart }
export { Catalogue as Catalogue }
export { Checkout as Checkout }
export { Contact as Contact }
export { Error as Error }
export { Home as Home }
export { OurRoasters as OurRoasters }
export { Product as Product }
export { TasteFinder as TasteFinder }
