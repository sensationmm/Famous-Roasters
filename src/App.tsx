import React from 'react'
import { I18nextProvider } from 'react-i18next'
import { BrowserRouter, Route, Routes } from 'react-router-dom'
import {
  About,
  Account,
  Cart,
  Catalogue,
  Checkout,
  Contact,
  Error,
  Home,
  OurRoasters,
  Product,
  TasteFinder,
} from 'src/views'

import { i18n } from './config'

function App() {
  return (
    <I18nextProvider i18n={i18n}>
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/about" element={<About />} />
          <Route path="/account" element={<Account />} />
          <Route path="/cart" element={<Cart />} />
          <Route path="/catalogue" element={<Catalogue />} />
          <Route path="/checkout" element={<Checkout />} />
          <Route path="/contact" element={<Contact />} />
          <Route path="/our-roasters" element={<OurRoasters />} />
          <Route path="/product/:id" element={<Product />} />
          <Route path="/taste-finder" element={<TasteFinder />} />
          <Route path="*" element={<Error />} />
        </Routes>
      </BrowserRouter>
    </I18nextProvider>
  )
}

export default App
