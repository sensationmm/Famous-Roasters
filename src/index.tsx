import './index.css'

import React from 'react'
import { createRoot } from 'react-dom/client'

import App from './App'

const container = document.getElementById('root') as Element
const root = createRoot(container)
if (process.env.NODE_ENV === 'development') {
  root.render(
    <React.StrictMode>
      <App />
    </React.StrictMode>,
  )
} else {
  root.render(<App />)
}
