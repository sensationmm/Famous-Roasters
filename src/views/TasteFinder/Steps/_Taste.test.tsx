import { render, waitFor } from '@testing-library/react'
import React from 'react'
import { I18nextProvider } from 'react-i18next'
import image1 from 'src/assets/images/tasteFinder/01-chocolate-leicht.webp'
import image2 from 'src/assets/images/tasteFinder/02-chocolate-mittel.webp'
import image3 from 'src/assets/images/tasteFinder/03-chocolate-hoch.webp'
import { i18n } from 'src/config'

import { IconName } from '../../../components'
import { TasteScreen, TasteScreenImageType } from '.'

const BitternessData = [
  {
    name: '1',
    image: image1,
    text: 'option 1',
    selectedText: 'is selected!',
  },
  {
    name: '2',
    image: image2,
    text: 'option 2',
    selectedText: 'is selected!',
  },
  {
    name: '3',
    image: image3,
    text: 'option 3',
    selectedText: 'is selected!',
  },
]

const BrewingData = [
  {
    name: 'french',
    iconName: IconName.FrenchPress,
    text: 'text 1',
  },
  {
    name: 'aeropress',
    iconName: IconName.Aeropress,
    text: 'text 2',
  },
  {
    name: 'v60',
    iconName: IconName.V60,
    text: 'text 3',
  },
  {
    name: 'filter',
    iconName: IconName.FilterkaffeeMachine,
    text: 'text 4',
  },
  {
    name: 'moka',
    iconName: IconName.Moka,
    text: 'text 5',
  },
  {
    name: 'chemex',
    iconName: IconName.Chemex,
    text: 'text 6',
  },
  {
    name: 'espresso',
    iconName: IconName.Espresso,
    text: 'text 7',
  },
  {
    name: 'vollautomat',
    iconName: IconName.VollautomatMachine,
    text: 'text 8',
  },
]

const ImagesWithoutImagesData = [
  {
    name: '1',
    iconName: IconName.Chemex,
    text: 'option 1',
    selectedText: 'is selected!',
  },
  {
    name: '2',
    iconName: IconName.Chemex,
    text: 'option 2',
    selectedText: 'is selected!',
  },
  {
    name: '3',
    iconName: IconName.Chemex,
    text: 'option 3',
    selectedText: 'is selected!',
  },
]

const IconsWithoutIconsData = [
  {
    name: '1',
    image: image1,
    text: 'option 1',
    selectedText: 'is selected!',
  },
  {
    name: '2',
    image: image1,
    text: 'option 2',
    selectedText: 'is selected!',
  },
  {
    name: '3',
    image: image1,
    text: 'option 3',
    selectedText: 'is selected!',
  },
]

global.alert = jest.fn()

describe('Taste screen dynamic partial view', () => {
  it('Renders correctly', async () => {
    const { container } = render(
      <I18nextProvider i18n={i18n}>
        <TasteScreen
          screenKey="bitterness"
          screenData={BitternessData}
          imageType={TasteScreenImageType.Image}
          currentData={[]}
          updateData={() => null}
        />
      </I18nextProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for not scrollable and image type icon', async () => {
    const { container } = render(
      <I18nextProvider i18n={i18n}>
        <TasteScreen
          screenKey="grindType"
          screenData={BrewingData}
          imageType={TasteScreenImageType.Icon}
          currentData={[]}
          updateData={() => null}
        />
      </I18nextProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Handles correctly image type without images', async () => {
    const { container } = render(
      <I18nextProvider i18n={i18n}>
        <TasteScreen
          screenKey="bitterness"
          screenData={ImagesWithoutImagesData}
          imageType={TasteScreenImageType.Image}
          scrollableImages={false}
          currentData={[]}
          updateData={() => null}
        />
      </I18nextProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Handles correctly icon type without icons', async () => {
    const { container } = render(
      <I18nextProvider i18n={i18n}>
        <TasteScreen
          screenKey="bitterness"
          screenData={IconsWithoutIconsData}
          imageType={TasteScreenImageType.Icon}
          scrollableImages={false}
          currentData={[]}
          updateData={() => null}
        />
      </I18nextProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Handles null selectedText', async () => {
    const { container } = render(
      <I18nextProvider i18n={i18n}>
        <TasteScreen
          screenKey="bitterness"
          screenData={[
            ...IconsWithoutIconsData,
            {
              name: '4',
              text: 'option 4',
              image: image1,
            },
          ]}
          imageType={TasteScreenImageType.Image}
          scrollableImages={true}
          currentData={[]}
          updateData={() => null}
        />
      </I18nextProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })
})
