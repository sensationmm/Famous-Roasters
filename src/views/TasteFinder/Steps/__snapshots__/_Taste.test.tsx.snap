// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Taste screen dynamic partial view Handles correctly icon type without icons 1`] = `
<div>
  <div
    class="flex-grow flex flex-col items-center bg-white md:justify-center"
  >
    <div
      class="p-6 md:w-5/6"
    >
      <h1
        class="flex md:text-2xl md:leading-8 xl:text-3xl xl:leading-10 text-xl leading-7 font-semibold"
      >
        Stell dir vor, dass verschiedene Sorten Schokolade vor dir auf dem Tisch liegen. Für welche würdest du dich entscheiden?
      </h1>
      <div
        class="flex md:hidden"
      >
        <button
          data-testid="drawer-trigger"
        >
          <div
            class="flex mt-4"
          >
            <svg
              class="mr-2.5"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                clip-rule="evenodd"
                d="M10 18.8a8.8 8.8 0 1 1 0-17.6 8.8 8.8 0 0 1 0 17.6ZM0 10C0 4.477 4.477 0 10 0s10 4.477 10 10-4.477 10-10 10S0 15.523 0 10Zm8.706-.725v2.607h1.83v-1.35a4.686 4.686 0 0 0 1.749-.435 3.152 3.152 0 0 0 1.253-1.07c.308-.466.462-1.03.462-1.691 0-.476-.088-.916-.264-1.32a2.89 2.89 0 0 0-.742-1.054c-.33-.31-.748-.548-1.253-.714C11.246 4.083 10.669 4 10.009 4c-.539 0-1.028.062-1.468.186-.428.114-.802.28-1.121.497-.319.207-.588.45-.808.729-.21.28-.368.574-.478.884-.1.3-.143.61-.132.931h1.847a1.61 1.61 0 0 1 .264-.853c.164-.248.4-.445.709-.59.308-.145.687-.217 1.138-.217.637 0 1.148.16 1.533.481.396.31.594.755.594 1.334 0 .6-.214 1.066-.643 1.397-.418.33-1.072.496-1.963.496h-.775Zm-.165 4.034V15h2.227v-1.691H8.54Z"
                fill="#212121"
                fill-rule="evenodd"
                stroke="#212121"
                stroke-width="0.1"
              />
            </svg>
            <span
              class="flex border-b-2 mb-1 text-sm leading-5"
              style="border-bottom-width: 1px;"
            >
              Was hat das mit Kaffee zu tun?
            </span>
          </div>
        </button>
      </div>
      <div
        class="hidden md:flex"
      >
        <button
          data-testid="dialog-trigger"
        >
          <div
            class="flex mt-4"
          >
            <svg
              class="mr-2.5"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                clip-rule="evenodd"
                d="M10 18.8a8.8 8.8 0 1 1 0-17.6 8.8 8.8 0 0 1 0 17.6ZM0 10C0 4.477 4.477 0 10 0s10 4.477 10 10-4.477 10-10 10S0 15.523 0 10Zm8.706-.725v2.607h1.83v-1.35a4.686 4.686 0 0 0 1.749-.435 3.152 3.152 0 0 0 1.253-1.07c.308-.466.462-1.03.462-1.691 0-.476-.088-.916-.264-1.32a2.89 2.89 0 0 0-.742-1.054c-.33-.31-.748-.548-1.253-.714C11.246 4.083 10.669 4 10.009 4c-.539 0-1.028.062-1.468.186-.428.114-.802.28-1.121.497-.319.207-.588.45-.808.729-.21.28-.368.574-.478.884-.1.3-.143.61-.132.931h1.847a1.61 1.61 0 0 1 .264-.853c.164-.248.4-.445.709-.59.308-.145.687-.217 1.138-.217.637 0 1.148.16 1.533.481.396.31.594.755.594 1.334 0 .6-.214 1.066-.643 1.397-.418.33-1.072.496-1.963.496h-.775Zm-.165 4.034V15h2.227v-1.691H8.54Z"
                fill="#212121"
                fill-rule="evenodd"
                stroke="#212121"
                stroke-width="0.1"
              />
            </svg>
            <span
              class="flex border-b-2 mb-1 text-sm leading-5"
              style="border-bottom-width: 1px;"
            >
              Was hat das mit Kaffee zu tun?
            </span>
          </div>
        </button>
      </div>
    </div>
    <div
      class="grid gap-x-4 w-full md:w-5/6 md:grid-rows-3 md:grid-cols-2 md:grid-flow-col"
    >
      <div
        class="scroll-auto border-b border-brand-grey-whisper"
      >
        <button
          class="flex w-full flex-row pl-6 py-4 items-center cursor-pointer"
          data-testid="icon-checkbox"
        >
          <div>
            <svg
              height="24"
              viewBox="0 0 16 16"
              width="24"
            >
              <path
                clip-rule="evenodd"
                d="M8,16c-4.4,0-8-3.6-8-8s3.6-8,8-8s8,3.6,8,8S12.4,16,8,16z M8,1C4.1,1,1,4.1,1,8s3.1,7,7,7s7-3.1,7-7S11.9,1,8,1z"
                fill="#D8D8D8"
                fill-rule="evenodd"
                stroke="#D8D8D8"
                stroke-width="0.1"
              />
            </svg>
          </div>
          <div
            class="flex grow text-left text-base leading-6 font-semibold"
          >
            option 1
          </div>
        </button>
      </div>
      <div
        class="scroll-auto border-b border-brand-grey-whisper"
      >
        <button
          class="flex w-full flex-row pl-6 py-4 items-center cursor-pointer"
          data-testid="icon-checkbox"
        >
          <div>
            <svg
              height="24"
              viewBox="0 0 16 16"
              width="24"
            >
              <path
                clip-rule="evenodd"
                d="M8,16c-4.4,0-8-3.6-8-8s3.6-8,8-8s8,3.6,8,8S12.4,16,8,16z M8,1C4.1,1,1,4.1,1,8s3.1,7,7,7s7-3.1,7-7S11.9,1,8,1z"
                fill="#D8D8D8"
                fill-rule="evenodd"
                stroke="#D8D8D8"
                stroke-width="0.1"
              />
            </svg>
          </div>
          <div
            class="flex grow text-left text-base leading-6 font-semibold"
          >
            option 2
          </div>
        </button>
      </div>
      <div
        class="scroll-auto border-b border-brand-grey-whisper md:border-0 scroll-auto"
      >
        <button
          class="flex w-full flex-row pl-6 py-4 items-center cursor-pointer"
          data-testid="icon-checkbox"
        >
          <div>
            <svg
              height="24"
              viewBox="0 0 16 16"
              width="24"
            >
              <path
                clip-rule="evenodd"
                d="M8,16c-4.4,0-8-3.6-8-8s3.6-8,8-8s8,3.6,8,8S12.4,16,8,16z M8,1C4.1,1,1,4.1,1,8s3.1,7,7,7s7-3.1,7-7S11.9,1,8,1z"
                fill="#D8D8D8"
                fill-rule="evenodd"
                stroke="#D8D8D8"
                stroke-width="0.1"
              />
            </svg>
          </div>
          <div
            class="flex grow text-left text-base leading-6 font-semibold"
          >
            option 3
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Taste screen dynamic partial view Handles correctly image type without images 1`] = `
<div>
  <div
    class="flex-grow flex flex-col items-center bg-white md:justify-center"
  >
    <div
      class="p-6 md:w-5/6"
    >
      <h1
        class="flex md:text-2xl md:leading-8 xl:text-3xl xl:leading-10 text-xl leading-7 font-semibold"
      >
        Stell dir vor, dass verschiedene Sorten Schokolade vor dir auf dem Tisch liegen. Für welche würdest du dich entscheiden?
      </h1>
      <div
        class="flex md:hidden"
      >
        <button
          data-testid="drawer-trigger"
        >
          <div
            class="flex mt-4"
          >
            <svg
              class="mr-2.5"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                clip-rule="evenodd"
                d="M10 18.8a8.8 8.8 0 1 1 0-17.6 8.8 8.8 0 0 1 0 17.6ZM0 10C0 4.477 4.477 0 10 0s10 4.477 10 10-4.477 10-10 10S0 15.523 0 10Zm8.706-.725v2.607h1.83v-1.35a4.686 4.686 0 0 0 1.749-.435 3.152 3.152 0 0 0 1.253-1.07c.308-.466.462-1.03.462-1.691 0-.476-.088-.916-.264-1.32a2.89 2.89 0 0 0-.742-1.054c-.33-.31-.748-.548-1.253-.714C11.246 4.083 10.669 4 10.009 4c-.539 0-1.028.062-1.468.186-.428.114-.802.28-1.121.497-.319.207-.588.45-.808.729-.21.28-.368.574-.478.884-.1.3-.143.61-.132.931h1.847a1.61 1.61 0 0 1 .264-.853c.164-.248.4-.445.709-.59.308-.145.687-.217 1.138-.217.637 0 1.148.16 1.533.481.396.31.594.755.594 1.334 0 .6-.214 1.066-.643 1.397-.418.33-1.072.496-1.963.496h-.775Zm-.165 4.034V15h2.227v-1.691H8.54Z"
                fill="#212121"
                fill-rule="evenodd"
                stroke="#212121"
                stroke-width="0.1"
              />
            </svg>
            <span
              class="flex border-b-2 mb-1 text-sm leading-5"
              style="border-bottom-width: 1px;"
            >
              Was hat das mit Kaffee zu tun?
            </span>
          </div>
        </button>
      </div>
      <div
        class="hidden md:flex"
      >
        <button
          data-testid="dialog-trigger"
        >
          <div
            class="flex mt-4"
          >
            <svg
              class="mr-2.5"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                clip-rule="evenodd"
                d="M10 18.8a8.8 8.8 0 1 1 0-17.6 8.8 8.8 0 0 1 0 17.6ZM0 10C0 4.477 4.477 0 10 0s10 4.477 10 10-4.477 10-10 10S0 15.523 0 10Zm8.706-.725v2.607h1.83v-1.35a4.686 4.686 0 0 0 1.749-.435 3.152 3.152 0 0 0 1.253-1.07c.308-.466.462-1.03.462-1.691 0-.476-.088-.916-.264-1.32a2.89 2.89 0 0 0-.742-1.054c-.33-.31-.748-.548-1.253-.714C11.246 4.083 10.669 4 10.009 4c-.539 0-1.028.062-1.468.186-.428.114-.802.28-1.121.497-.319.207-.588.45-.808.729-.21.28-.368.574-.478.884-.1.3-.143.61-.132.931h1.847a1.61 1.61 0 0 1 .264-.853c.164-.248.4-.445.709-.59.308-.145.687-.217 1.138-.217.637 0 1.148.16 1.533.481.396.31.594.755.594 1.334 0 .6-.214 1.066-.643 1.397-.418.33-1.072.496-1.963.496h-.775Zm-.165 4.034V15h2.227v-1.691H8.54Z"
                fill="#212121"
                fill-rule="evenodd"
                stroke="#212121"
                stroke-width="0.1"
              />
            </svg>
            <span
              class="flex border-b-2 mb-1 text-sm leading-5"
              style="border-bottom-width: 1px;"
            >
              Was hat das mit Kaffee zu tun?
            </span>
          </div>
        </button>
      </div>
    </div>
    <div
      class="grid gap-x-4 w-full md:w-5/6 md:grid-rows-3 md:grid-cols-2 md:grid-flow-col"
    />
  </div>
</div>
`;

exports[`Taste screen dynamic partial view Renders correctly 1`] = `
<div>
  <div
    class="flex-grow flex flex-col items-center bg-white md:justify-center"
  >
    <div
      class="p-6 md:w-5/6"
    >
      <h1
        class="flex md:text-2xl md:leading-8 xl:text-3xl xl:leading-10 text-xl leading-7 font-semibold"
      >
        Stell dir vor, dass verschiedene Sorten Schokolade vor dir auf dem Tisch liegen. Für welche würdest du dich entscheiden?
      </h1>
      <div
        class="flex md:hidden"
      >
        <button
          data-testid="drawer-trigger"
        >
          <div
            class="flex mt-4"
          >
            <svg
              class="mr-2.5"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                clip-rule="evenodd"
                d="M10 18.8a8.8 8.8 0 1 1 0-17.6 8.8 8.8 0 0 1 0 17.6ZM0 10C0 4.477 4.477 0 10 0s10 4.477 10 10-4.477 10-10 10S0 15.523 0 10Zm8.706-.725v2.607h1.83v-1.35a4.686 4.686 0 0 0 1.749-.435 3.152 3.152 0 0 0 1.253-1.07c.308-.466.462-1.03.462-1.691 0-.476-.088-.916-.264-1.32a2.89 2.89 0 0 0-.742-1.054c-.33-.31-.748-.548-1.253-.714C11.246 4.083 10.669 4 10.009 4c-.539 0-1.028.062-1.468.186-.428.114-.802.28-1.121.497-.319.207-.588.45-.808.729-.21.28-.368.574-.478.884-.1.3-.143.61-.132.931h1.847a1.61 1.61 0 0 1 .264-.853c.164-.248.4-.445.709-.59.308-.145.687-.217 1.138-.217.637 0 1.148.16 1.533.481.396.31.594.755.594 1.334 0 .6-.214 1.066-.643 1.397-.418.33-1.072.496-1.963.496h-.775Zm-.165 4.034V15h2.227v-1.691H8.54Z"
                fill="#212121"
                fill-rule="evenodd"
                stroke="#212121"
                stroke-width="0.1"
              />
            </svg>
            <span
              class="flex border-b-2 mb-1 text-sm leading-5"
              style="border-bottom-width: 1px;"
            >
              Was hat das mit Kaffee zu tun?
            </span>
          </div>
        </button>
      </div>
      <div
        class="hidden md:flex"
      >
        <button
          data-testid="dialog-trigger"
        >
          <div
            class="flex mt-4"
          >
            <svg
              class="mr-2.5"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                clip-rule="evenodd"
                d="M10 18.8a8.8 8.8 0 1 1 0-17.6 8.8 8.8 0 0 1 0 17.6ZM0 10C0 4.477 4.477 0 10 0s10 4.477 10 10-4.477 10-10 10S0 15.523 0 10Zm8.706-.725v2.607h1.83v-1.35a4.686 4.686 0 0 0 1.749-.435 3.152 3.152 0 0 0 1.253-1.07c.308-.466.462-1.03.462-1.691 0-.476-.088-.916-.264-1.32a2.89 2.89 0 0 0-.742-1.054c-.33-.31-.748-.548-1.253-.714C11.246 4.083 10.669 4 10.009 4c-.539 0-1.028.062-1.468.186-.428.114-.802.28-1.121.497-.319.207-.588.45-.808.729-.21.28-.368.574-.478.884-.1.3-.143.61-.132.931h1.847a1.61 1.61 0 0 1 .264-.853c.164-.248.4-.445.709-.59.308-.145.687-.217 1.138-.217.637 0 1.148.16 1.533.481.396.31.594.755.594 1.334 0 .6-.214 1.066-.643 1.397-.418.33-1.072.496-1.963.496h-.775Zm-.165 4.034V15h2.227v-1.691H8.54Z"
                fill="#212121"
                fill-rule="evenodd"
                stroke="#212121"
                stroke-width="0.1"
              />
            </svg>
            <span
              class="flex border-b-2 mb-1 text-sm leading-5"
              style="border-bottom-width: 1px;"
            >
              Was hat das mit Kaffee zu tun?
            </span>
          </div>
        </button>
      </div>
    </div>
    <div
      class="relative flex w-full overflow-x-auto gap-x-10 snap-x xl:justify-center"
    >
      <div
        class="w-1/12 pr-12 -mx-2.5 h-80 relative shrink-0 snap-start scroll-auto"
      />
      <button
        class="w-64 h-80 mt-12 -top-4 relative cursor-pointer shrink-0 snap-center scroll-auto"
        data-testid="image-checkbox"
      >
        <div
          class="w-64 h-64 top-0 absolute rounded-full bg-brand-grey-whisper"
        />
        <img
          alt=""
          class="absolute -top-8"
          src="01-chocolate-leicht.webp"
        />
        <div
          class="absolute w-full h-32 text-center -bottom-4"
        >
          <div
            class="font-syne font-normal text-2xl leading-8 font-semibold"
          >
            option 1
          </div>
        </div>
      </button>
      <button
        class="w-64 h-80 mt-12 -top-4 relative cursor-pointer shrink-0 snap-center scroll-auto"
        data-testid="image-checkbox"
      >
        <div
          class="w-64 h-64 top-0 absolute rounded-full bg-brand-grey-whisper"
        />
        <img
          alt=""
          class="absolute -top-8"
          src="02-chocolate-mittel.webp"
        />
        <div
          class="absolute w-full h-32 text-center -bottom-4"
        >
          <div
            class="font-syne font-normal text-2xl leading-8 font-semibold"
          >
            option 2
          </div>
        </div>
      </button>
      <button
        class="w-64 h-80 mt-12 -top-4 relative cursor-pointer shrink-0 snap-center scroll-auto"
        data-testid="image-checkbox"
      >
        <div
          class="w-64 h-64 top-0 absolute rounded-full bg-brand-grey-whisper"
        />
        <img
          alt=""
          class="absolute -top-8"
          src="03-chocolate-hoch.webp"
        />
        <div
          class="absolute w-full h-32 text-center -bottom-4"
        >
          <div
            class="font-syne font-normal text-2xl leading-8 font-semibold"
          >
            option 3
          </div>
        </div>
      </button>
      <div
        class="w-1/12 pl-12 -mx-2.5 h-80 relative shrink-0 snap-start scroll-auto"
      />
    </div>
  </div>
</div>
`;

exports[`Taste screen dynamic partial view Renders correctly for not scrollable and image type icon 1`] = `
<div>
  <div
    class="flex-grow flex flex-col items-center bg-white md:justify-center"
  >
    <div
      class="p-6 md:w-5/6"
    >
      <h1
        class="flex md:text-2xl md:leading-8 xl:text-3xl xl:leading-10 text-xl leading-7 font-semibold"
      >
        Jeder weiß: der Tag beginnt nach der ersten Tasse Kaffee. Auf welche Art würdest du deinen Kaffee brühen?
      </h1>
      <div
        class="flex md:hidden"
      >
        <button
          data-testid="drawer-trigger"
        >
          <div
            class="flex mt-4"
          >
            <svg
              class="mr-2.5"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                clip-rule="evenodd"
                d="M10 18.8a8.8 8.8 0 1 1 0-17.6 8.8 8.8 0 0 1 0 17.6ZM0 10C0 4.477 4.477 0 10 0s10 4.477 10 10-4.477 10-10 10S0 15.523 0 10Zm8.706-.725v2.607h1.83v-1.35a4.686 4.686 0 0 0 1.749-.435 3.152 3.152 0 0 0 1.253-1.07c.308-.466.462-1.03.462-1.691 0-.476-.088-.916-.264-1.32a2.89 2.89 0 0 0-.742-1.054c-.33-.31-.748-.548-1.253-.714C11.246 4.083 10.669 4 10.009 4c-.539 0-1.028.062-1.468.186-.428.114-.802.28-1.121.497-.319.207-.588.45-.808.729-.21.28-.368.574-.478.884-.1.3-.143.61-.132.931h1.847a1.61 1.61 0 0 1 .264-.853c.164-.248.4-.445.709-.59.308-.145.687-.217 1.138-.217.637 0 1.148.16 1.533.481.396.31.594.755.594 1.334 0 .6-.214 1.066-.643 1.397-.418.33-1.072.496-1.963.496h-.775Zm-.165 4.034V15h2.227v-1.691H8.54Z"
                fill="#212121"
                fill-rule="evenodd"
                stroke="#212121"
                stroke-width="0.1"
              />
            </svg>
            <span
              class="flex border-b-2 mb-1 text-sm leading-5"
              style="border-bottom-width: 1px;"
            >
              Was hat das mit Kaffee zu tun?
            </span>
          </div>
        </button>
      </div>
      <div
        class="hidden md:flex"
      >
        <button
          data-testid="dialog-trigger"
        >
          <div
            class="flex mt-4"
          >
            <svg
              class="mr-2.5"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                clip-rule="evenodd"
                d="M10 18.8a8.8 8.8 0 1 1 0-17.6 8.8 8.8 0 0 1 0 17.6ZM0 10C0 4.477 4.477 0 10 0s10 4.477 10 10-4.477 10-10 10S0 15.523 0 10Zm8.706-.725v2.607h1.83v-1.35a4.686 4.686 0 0 0 1.749-.435 3.152 3.152 0 0 0 1.253-1.07c.308-.466.462-1.03.462-1.691 0-.476-.088-.916-.264-1.32a2.89 2.89 0 0 0-.742-1.054c-.33-.31-.748-.548-1.253-.714C11.246 4.083 10.669 4 10.009 4c-.539 0-1.028.062-1.468.186-.428.114-.802.28-1.121.497-.319.207-.588.45-.808.729-.21.28-.368.574-.478.884-.1.3-.143.61-.132.931h1.847a1.61 1.61 0 0 1 .264-.853c.164-.248.4-.445.709-.59.308-.145.687-.217 1.138-.217.637 0 1.148.16 1.533.481.396.31.594.755.594 1.334 0 .6-.214 1.066-.643 1.397-.418.33-1.072.496-1.963.496h-.775Zm-.165 4.034V15h2.227v-1.691H8.54Z"
                fill="#212121"
                fill-rule="evenodd"
                stroke="#212121"
                stroke-width="0.1"
              />
            </svg>
            <span
              class="flex border-b-2 mb-1 text-sm leading-5"
              style="border-bottom-width: 1px;"
            >
              Was hat das mit Kaffee zu tun?
            </span>
          </div>
        </button>
      </div>
    </div>
    <div
      class="relative flex w-full overflow-x-auto gap-x-10 snap-x xl:justify-center"
    >
      <div
        class="w-1/12 pr-12 -mx-2.5 h-80 relative shrink-0 snap-start scroll-auto"
      />
      <div
        class="w-1/12 pl-12 -mx-2.5 h-80 relative shrink-0 snap-start scroll-auto"
      />
    </div>
  </div>
</div>
`;
